core/src/main/java/org/bitcoinj/script/ScriptBuilder.java

```diff
@@ -478,6 +478,16 @@ public static Script createP2WPKHOutputScript(ECKey key) {
         return createP2WPKHOutputScript(key.getPubKeyHash());
     }
 
+    /**
+     * Creates a scriptPubKey that sends to script calculated by the given key. Read
+     * <a href="https://github.com/bitcoin/bips/blob/master/bip-0016.mediawiki">BIP 16</a> to learn more about this
+     * kind of script.
+     */
+    public static Script createP2SHOutputScript(ECKey key) {
+        final Script redeemScript = new ScriptBuilder().smallNum(0).data(key.getPubKeyHash()).build();
```

Comment:
The line creating the redeemScript should probably be in a separate method, similar to the existing `createRedeemScript` (for multisig) because we'll need that code for redeeming these P2SH-P2WPKH outputs.